src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -4078,4 +4078,155 @@ BOOST_AUTO_TEST_CASE(GradientBatchNormWithMiniBatchesTest)
   BOOST_REQUIRE(pass);
 }
 
+/**
+ * Simple Multihead Attention test.
+ */
+BOOST_AUTO_TEST_CASE(SimpleMultiheadAttentionTest)
+{
+  size_t tLen = 50;
+  size_t sLen = tLen;
+  size_t embedDim = 512;
+  size_t numHeads = 16;
+  size_t bsz = 64;
+  double INF = std::numeric_limits<double>::max();
+
+  arma::mat query = 0.1 * arma::randu(embedDim * tLen, bsz);
+  arma::mat output;
+
+  arma::mat attnMask = arma::zeros(tLen, sLen);
+  for (size_t i = 0; i < tLen; ++i)
+  {
+    for (size_t j = 0; j < sLen; ++j)
+    {
+      if (i < j)
+        attnMask(i, j) = -INF;
+    }
+  }
+
+  arma::mat keyPaddingMask = arma::zeros(1, sLen);
+  keyPaddingMask(sLen - 1) = -INF;
```

Comment:
I think it's better to use `lowest()` instead of `-max()`.
```suggestion
  keyPaddingMask(sLen - 1) = std::numeric_limits<double>::lowest();
```